tests/integration/test_charm.py

```diff
@@ -8,6 +9,35 @@
 from selenium.webdriver.firefox.options import Options
 from selenium.webdriver.support.ui import WebDriverWait
 
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test):
+    my_charm = await ops_test.build_charm(".")
+    image_path = METADATA["resources"]["oci-image"]["upstream-source"]
+    resources = {"oci-image": image_path}
```

Comment:
This could probably just be inlined into the call to `ops_test.model.deploy(...)`